In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import datetime as dt
import seaborn as sns
import re
from datetime import date, timedelta
import scipy.stats as stats
import math
# set the graphs to show in the jupyter notebook
%matplotlib inline

# set seabor graphs to a better style
sns.set(style="ticks")

In [3]:
loan = pd.read_csv('LoansData.csv')

# BUSINESS PROBLEM - 1 

## (a) Intrest rate is varied for different loan amounts (Less intrest charged for high loan amounts)

In [5]:
loan['Interest.Rate'] = loan['Interest.Rate'].str.replace('%','').astype(float)

In [21]:
loan.corr(method ='pearson').iloc[1,2]

0.3374540043675969

There is no correlation between Interest rates and loan amounts.

## (b)  Loan length is directly effecting intrest rate.

In [ ]:
loan['Loan.Length'] = loan['Loan.Length'].str.replace('months','').astype(int)

In [19]:
loan.corr(method ='pearson').iloc[2,3]

0.424219602761327

There is very mild correlation between loan length and interest rate, so loan length is not directly effecting interest rate.

## (c)  Interest rate varies for different purpose of loans.

In [32]:
debt_consolidation = loan.loc[(loan['Loan.Purpose'] == 'debt_consolidation'),'Interest.Rate']
credit_card = loan.loc[(loan['Loan.Purpose'] == 'credit_card'),'Interest.Rate']
other = loan.loc[(loan['Loan.Purpose'] == 'other'),'Interest.Rate']
moving = loan.loc[(loan['Loan.Purpose'] == 'moving'),'Interest.Rate']
car = loan.loc[(loan['Loan.Purpose'] == 'car'),'Interest.Rate']
vacation = loan.loc[(loan['Loan.Purpose'] == 'vacation'),'Interest.Rate']
home_improvement = loan.loc[(loan['Loan.Purpose'] == 'home_improvement'),'Interest.Rate']
house = loan.loc[(loan['Loan.Purpose'] == 'house'),'Interest.Rate']
major_purchase = loan.loc[(loan['Loan.Purpose'] == 'major_purchase'),'Interest.Rate']
educational = loan.loc[(loan['Loan.Purpose'] == 'educational'),'Interest.Rate']
medical = loan.loc[(loan['Loan.Purpose'] == 'medical'),'Interest.Rate']
wedding= loan.loc[(loan['Loan.Purpose'] == 'wedding'),'Interest.Rate']
small_business = loan.loc[(loan['Loan.Purpose'] == 'small_business'),'Interest.Rate']
renewable_energy = loan.loc[(loan['Loan.Purpose'] == 'renewable_energy'),'Interest.Rate']

In [30]:
loan['Loan.Purpose'].unique()

array(['debt_consolidation', 'credit_card', 'other', 'moving', 'car',
       'vacation', 'home_improvement', 'house', 'major_purchase',
       'educational', 'medical', 'wedding', 'small_business',
       'renewable_energy'], dtype=object)

In [68]:
credit_card = credit_card.dropna()
wedding = wedding.dropna()


In [69]:
print('Mean for debt_consolidation',round(debt_consolidation.mean(),2))
print('Mean for credit_card',round(credit_card.mean(),2))
print('Mean for other',round(other.mean(),2))
print('Mean for moving',round(moving.mean(),2))
print('Mean for car',round(car.mean(),2))
print('Mean for vacation',round(vacation.mean(),2))
print('Mean for home_improvement',round(home_improvement.mean(),2))
print('Mean for house',round(house.mean(),2))
print('Mean for major_purchase',round(major_purchase.mean(),2))
print('Mean for educational',round(educational.mean(),2))
print('Mean for medical',round(medical.mean(),2))
print('Mean for wedding',round(wedding.mean(),2))
print('Mean for small_business',round(small_business.mean(),2))
print('Mean for renewable_energy',round(renewable_energy.mean(),2))

Mean for debt_consolidation 13.59
Mean for credit_card 13.05
Mean for other 13.16
Mean for moving 13.62
Mean for car 11.11
Mean for vacation 11.97
Mean for home_improvement 11.59
Mean for house 13.45
Mean for major_purchase 10.8
Mean for educational 11.01
Mean for medical 11.65
Mean for wedding 12.05
Mean for small_business 12.84
Mean for renewable_energy 9.88


In [71]:
stats.f_oneway(debt_consolidation,credit_card,other,moving,car,vacation,home_improvement,house,major_purchase,educational,medical,wedding,small_business,renewable_energy)

F_onewayResult(statistic=7.481404351302254, pvalue=1.1686298123677374e-14)

pvalue(1.16) is greater than significance level(0.05) so H0(null hypothesis) will apply i.e., interest rate does not vary with different purpose of loans.

## (d) There is relationship between FICO scores and Home Ownership. It means that, People with owning home will have high FICO scores.

In [75]:
loan[['Fico_Low', 'Fico_High']] = loan['FICO.Range'].str.split('-',expand=True)

In [77]:
loan.Fico_Low = loan.Fico_Low.astype('float64')
loan.Fico_High = loan.Fico_High.astype('float64')

In [80]:
loan['Fico_Avg'] = (loan.Fico_Low+loan.Fico_High)/2

In [82]:
loan['Home.Ownership'].unique()

array(['MORTGAGE', 'RENT', 'OWN', 'OTHER', 'NONE', nan], dtype=object)

In [84]:
mortgage = loan.loc[(loan['Home.Ownership'] == 'MORTGAGE'),'Fico_Avg']
rent = loan.loc[(loan['Home.Ownership'] == 'RENT'),'Fico_Avg']
own = loan.loc[(loan['Home.Ownership'] == 'OWN'),'Fico_Avg']
other = loan.loc[(loan['Home.Ownership'] == 'OTHER'),'Fico_Avg']
none= loan.loc[(loan['Home.Ownership'] == 'NONE'),'Fico_Avg']

In [86]:
print('Mean for mortgage',round(mortgage.mean(),2))
print('Mean for rent',round(rent.mean(),2))
print('Mean for own',round(own.mean(),2))
print('Mean for other',round(other.mean(),2))
print('Mean for none',round(none.mean(),2))

Mean for mortgage 713.7
Mean for rent 702.06
Mean for own 708.55
Mean for other 674.0
Mean for none 802.0


In [91]:
mortgage = mortgage.dropna()
rent = rent.dropna()
own = own.dropna()
other = other.dropna()
none = none.dropna()

In [92]:
stats.f_oneway(mortgage,rent,own,other,none)

F_onewayResult(statistic=19.334591709631045, pvalue=1.114378111228758e-15)

p value is very less than significance level(0.05) so Ha(alternate hypothesis) will apply i.e., fico scores are different for different home ownership.

# BUSINESS PROBLEM - 2

In [5]:
price = pd.read_csv('Price_Quotes.csv')

In [22]:
print(round(price.Barry_Price.mean(),2))
print(round(price.Mary_Price.mean(),2))

124.33
114.75


In [23]:
stats.ttest_rel(a=price.Barry_Price, b=price.Mary_Price)

Ttest_relResult(statistic=2.521376510892349, pvalue=0.02840588045242053)

-t val: 2.52
-Significance Level: 0.05 (5%)
-P value: 0.028
-p val < signi value
-Accepting the Ha
-Ha: 124.33 != 114.75

Analysis : There is a difference between average price quotes provided by Mary and Barry.

# BUSINESS PROBLEM - 3

## Effect of reengineering effort on employee turnover. 

In [6]:
data = pd.read_csv('Treatment_Facility.csv')

In [7]:
data

Month Reengineer  Employee_Turnover       VAR4        VAR5
0       1      Prior             0.0000  24.390244   42.682927
1       2      Prior             6.0606  19.354839   25.806452
2       3      Prior            12.1212  35.087719  146.198830
3       4      Prior             3.3333  18.404908  110.429448
4       5      Prior            12.9032  17.964072   23.952096
5       6      Prior             9.6774  41.176471   47.058824
6       7      Prior            11.7647  13.422819    0.000000
7       8      Prior            11.4286  31.250000   25.000000
8       9      Prior            23.0769  17.241379  132.183908
9      10      Prior            15.0000  16.574586   16.574586
10     11      Prior            10.0000   5.917160   88.757396
11     12      Prior            26.0870  21.164021   31.746032
12     13      Prior            11.1111   5.076142   10.152284
13     14       Post            26.6667   9.708738   29.126214
14     15       Post            17.9487   4.201681   21.008403
15     16       Post            12.8205  13.043478   34.782609
16     17       Post            15.7895   9.950249   19.900498
17     18       Post             0.0000   9.950249    9.950249
18     19       Post            30.7692   8.510638   25.531915
19     20       Post            26.8293   9.259259   23.148148

In [30]:
prior = data.loc[(data.Reengineer == 'Prior'),'Employee_Turnover']
post = data.loc[(data.Reengineer == 'Post'),'Employee_Turnover']

In [34]:
print('Mean for Prior',round(prior.mean(),2))
print('Mean for Post',round(post.mean(),2))

Mean for Prior 11.74
Mean for Post 18.69


In [35]:
stats.ttest_ind(a= prior, b= post, equal_var=False)

Ttest_indResult(statistic=-1.5653912078421088, pvalue=0.15207128913702453)

-t val: -1.56
-Significance Level: 0.05 (5%)
-P value: 0.15 (HIGH)
-p val > signi value
-Accepting the H0
-H0: 11.74 == 18.69

Analysis : There is no statistically significant difference on employee turnover after reengineering effort.

##  Effect of reengineering effort on residents who are temporarily removed from the facility.

In [37]:
prior_r = data.loc[(data.Reengineer == 'Prior'),'VAR4']
post_r = data.loc[(data.Reengineer == 'Post'),'VAR4']

In [38]:
print('Mean for Prior',round(prior_r.mean(),2))
print('Mean for Post',round(post_r.mean(),2))

Mean for Prior 20.54
Mean for Post 9.23


In [39]:
stats.ttest_ind(a= prior_r, b= post_r, equal_var=False)

Ttest_indResult(statistic=3.6903825303098836, pvalue=0.0022744273676772084)

-t val: 3.69
-Significance Level: 0.05 (5%)
-P value: 0.002 (HIGH)
-p val < signi value
-Accepting the Ha
-Ha: 20.54 != 9.23

Analysis : Number of residents removed from the facility decreased after the reengineering effort.

## Effect of  reengineering effort on critical incident.

In [40]:
prior_c = data.loc[(data.Reengineer == 'Prior'),'VAR5']
post_c = data.loc[(data.Reengineer == 'Post'),'VAR5']

In [41]:
print('Mean for Prior',round(prior_c.mean(),2))
print('Mean for Post',round(post_c.mean(),2))

Mean for Prior 53.89
Mean for Post 23.35


In [42]:
stats.ttest_ind(a= prior_c, b= post_c, equal_var=False)

Ttest_indResult(statistic=2.209006531430452, pvalue=0.045565519570193176)

-t val: 2.209
-Significance Level: 0.05 (5%)
-P value: 0.045 (HIGH)
-p val < signi value
-Accepting the Ha
-Ha: 53.89 != 23.35

Analysis : Number of critical incidents decreased after the reengineering effort.

# BUSINESS PROBLEM - 4

In [43]:
priority = pd.read_csv('Priority_Assessment.csv')

In [44]:
priority

Days Priority
0     3.3     High
1     7.9   Medium
2     0.3     High
3     0.7   Medium
4     8.6   Medium
..    ...      ...
637   2.5      Low
638   0.3     High
639   0.3   Medium
640   1.3   Medium
641   0.5     High

[642 rows x 2 columns]

In [48]:
low = priority.loc[(priority.Priority == 'Low'),'Days']
Medium = priority.loc[(priority.Priority == 'Medium'),'Days']
High = priority.loc[(priority.Priority == 'High'),'Days']

In [49]:
print('Mean for low priority',round(low.mean(),2))
print('Mean for Medium priority',round(Medium.mean(),2))
print('Mean for High priority',round(High.mean(),2))

Mean for low priority 4.23
Mean for Medium priority 2.5
Mean for High priority 3.02


### First we check for low and medium priority

In [52]:
stats.ttest_ind(a= low, b= Medium, equal_var=False)

Ttest_indResult(statistic=2.0844370107136743, pvalue=0.039955774952838315)

-t val: 2.08
-Significance Level: 0.05 (5%)
-P value: 0.039 
-p val < signi value
-Accepting the Ha
-Ha: 4.23 != 2.5

Analysis : Medium priority jobs are done in less days as comapred to low priority jobs.

### Now we check for medium and high priority 

In [55]:
stats.ttest_ind(a= Medium, b= High, equal_var=False)

Ttest_indResult(statistic=-0.9780778483608645, pvalue=0.32845956565961953)

-t val: -0.97
-Significance Level: 0.05 (5%)
-P value: 0.328 
-p val > signi value
-Accepting the H0
-H0: 2.5 == 3.02

Analysis : Medium priority jobs are statistically done in same number of days as high priority jobs.

# BUSINESS PROBLEM - 5

In [56]:
film = pd.read_csv('Films.csv')

### - What is the overall level of customer satisfaction.

In [60]:
round(film.Overall.mean(),0)

2.0

- Overall customer satisfaction is Good

### - What factors are linked to satisfaction. 

(1) Signage
(2) Parking
(3) Clean

### - What is the demographic profile of Film on the Rocks patrons?

In [71]:
round(film.loc[(film['Movie'] == 'Ferris Buellers Day Off'),'Age'].mean(),0)

2.0

- Demographic profile for Ferris Buellers Day Off is age between 13-30.

In [72]:
round(film.loc[(film['Movie'] == 'Old School'),'Age'].mean(),0)

2.0

- Demographic profile for Old School is age between 13-30.

In [75]:
round(film.loc[(film['Movie'] == 'Willy Wonka'),'Age'].mean(),0)

2.0

- Demographic profile for Old School is age between 13-30.

### - In what media outlet(s) should the film series be advertised?

In [87]:
film.Hear_About.mode()

0    5
dtype: object